app/models/solid_queue/job/executable.rb

```diff
@@ -66,6 +66,8 @@ def prepare_for_execution
 
       def dispatch
         if acquire_concurrency_lock then ready
+        elsif discard_concurrent?
+          discard
```

Comment:
I don't think this is the way 🤔  We're in the middle of a transaction here, and the job hasn't even been committed to the DB. It makes no sense to delete a record in the same transaction you're creating it. It'd make sense to roll that transaction back instead.

Reply:
Perhaps I don't fully understand how we're getting into this code path, but from my investigation, it looks as though there isn't an open transaction at this time.

I tried the following and running the test suite and didn't hit any open transactions.

```
raise "open transactions" if ApplicationRecord.connection.open_transactions.positive?
```

I still believe we'd want to discard here. Let me know your thoughts.